### ISSUE TITLE
Figma component and icon metadata updates

### ISSUE THREAD

**Issue by bhoppers2008** [jpmorganchase/salt-ds#4094]
Batch update to be made to all Salt Next components:

1. Add "Salt DS, Salt Next theme" to **Library** row
2. Remove **Theme** row from all components

Batch update to be made to all Salt Icons in the Icon library:

- Library = Salt DS
- Name = [Icon name]
- Dev URL = Icon docs
- Design URL = Icon library

('Style' prop hopefully comes for free).

**Comment by origami-z** [comment]
> 1. Add "Salt DS, Salt Next theme" to **Library** row

@bhoppers2008 what will happen after legacy theme is connected ? ref- #3846

**Comment by mark-tate** [comment]
Frappe Goal: released by EOS

**Comment by origami-z** [comment]
Code for update Icon: 

```
for (const selection of figma.currentPage.selection) {
    selection.setSharedPluginData(
        "JPMC.CIB.DXD",
        "COMPONENT.INFO",
        JSON.stringify({
            Name: selection.parent.name, 
            Library: 'Salt DS',
            "Dev URL": "https://www.saltdesignsystem.com/salt/components/icon/",
            "Design URL": "https://www.figma.com/design/9laF7Zn75aUMNsqSqRgAuX/Salt-Icons?node-id=219-1627&t=llYU7Wmy8aHBqjVE-4"
        })
    );
}
```

**Comment by origami-z** [comment]
To update components, page by page

```
for (const page of figma.root.children) {
  await page.loadAsync()
  // console.log(`Page ${page.name} has ${page.children.length} children`)

  const components = figma.currentPage.findAllWithCriteria({
    types: ['COMPONENT']
  })
  
  for (const comp of components) {
    const pluginData = comp.getSharedPluginData(
      "JPMC.CIB.DXD",
      "COMPONENT.INFO"
    );
    if (pluginData) {
      const data = JSON.parse(pluginData);
      delete data.Theme;
      data.Library = 'Salt DS, Salt Next theme';
      comp.setSharedPluginData(
        "JPMC.CIB.DXD",
        "COMPONENT.INFO",
        JSON.stringify(data)
    );
    }
  }
  break;
}
```

**Comment by origami-z** [comment]
Sep 23 - Scripts above are ready to update both libraries. Need to work out when / which branch to apply, so avoid conflict with other on going changes.

**Comment by mark-tate** [comment]
Galao Goal: complete before release

**Comment by origami-z** [comment]
meeting set up on Friday to do this before release

**Comment by origami-z** [comment]
Updated in below branches. To test, launch Meta data plugin, check data is according to this issue's description

In the review and merge window, you should see each components only having "plugin data updated", not anything else.

- Component (From A-Tooltip, not Patterns) : https://www.figma.com/design/ChsbbO7pLomT4F5H6tQyLP/branch/ydcLhmDJI7fYBNv8lXGjlt/Salt-(Next)-Components-%26-Patterns?m=auto&t=hTyHsd9Tj95zHQLB-1
- Icons: https://www.figma.com/design/9laF7Zn75aUMNsqSqRgAuX/branch/co4JlCb1md612g31GRcnXc/Salt-Icons?m=auto&node-id=0-1&t=hOfwTMMcMwkNcBLi-1

**Comment by pseys** [comment]
> Updated in below branches. To test, launch Meta data plugin, check data is according to this issue's description
> 
> In the review and merge window, you should see each components only having "plugin data updated", not anything else.
> 
> * Component (From A-Tooltip, not Patterns) : https://www.figma.com/design/ChsbbO7pLomT4F5H6tQyLP/branch/ydcLhmDJI7fYBNv8lXGjlt/Salt-(Next)-Components-%26-Patterns?m=auto&t=hTyHsd9Tj95zHQLB-1
> * Icons: https://www.figma.com/design/9laF7Zn75aUMNsqSqRgAuX/branch/co4JlCb1md612g31GRcnXc/Salt-Icons?m=auto&node-id=0-1&t=hOfwTMMcMwkNcBLi-1

The icon library change appears to be working for me

**Comment by origami-z** [comment]
Changes all done. Will be released in #4185

**Comment by origami-z** [comment]
Code to add dev links to dev mode

```
const components = figma.currentPage.findAllWithCriteria({
  types: ["COMPONENT"],
});

for (const comp of components) {
  const pluginData = comp.getSharedPluginData("JPMC.CIB.DXD", "COMPONENT.INFO");
  if (pluginData) {
    const data = JSON.parse(pluginData);
    const devUrl = data['Dev URL'];
    if (devUrl) {
      console.log(devUrl);
      await comp.addDevResourceAsync(devUrl)
      console.log("added", comp.name)
    }
  }
}

```
